analyzer/src/main/kotlin/managers/SpdxTransitiveDocument.kt

```diff
@@ -0,0 +1,329 @@
+import org.ossreviewtoolkit.utils.spdx.model.SpdxExternalDocumentReference
+import org.ossreviewtoolkit.utils.spdx.model.SpdxPackage
+
+/**
+ * A data class storing information about a root SPDX document and all the documents referenced by it.
+ *
+ * This class is used by [SpdxDocumentFile] to get a combined view on all packages and their relations defined in a set
+ * of SPDX files.
+ */
+internal data class SpdxTransitiveDocument(
+    /**
+     * The root document. This is the starting point, from which all external references have been traversed.
+     */
+    val rootDocument: SpdxDocument,
+
+    /**
+     * The name of the package manager that uses this document. This is mainly used to fill the source in generated
+     * [OrtIssue]s.
+     */
+    val managerName: String,
+
+    /**
+     * Holds a map with all [SpdxDocument]s that are referenced directly or indirectly from the root document, using
+     * the external reference objects as keys.
+     */
+    val referencedDocuments: Map<SpdxExternalDocumentReference, SpdxDocument>,
+
+    /**
+     * A map allowing direct access to the packages declared in one of the contained documents. For the packages of the
+     * root document, the key is the package's identifiers. For other packages, the identifier needs to be prefixed
+     * with the document identifier of the reference.
+     */
+    private val packagesById: Map<String, SpdxPackage>,
+
+    /**
+     * A map storing issues that were encountered when resolving external document references. These issues are also
+     * assigned to [SpdxPackage]s defined in the corresponding external documents.
+     */
+    private val issuesByReferenceId: Map<String, OrtIssue>
+) {
+    companion object {
+        fun load(loader: SpdxDocumentLoader, rootDocumentFile: File, managerName: String): SpdxTransitiveDocument {
+            val rootDocument = loader.load(rootDocumentFile)
+
+            val references = mutableMapOf<SpdxExternalDocumentReference, SpdxDocument>()
+            val issues = mutableMapOf<String, OrtIssue>()
+            resolveTransitiveReferences(
+                loader,
+                managerName,
+                rootDocument,
+                rootDocumentFile.toURI(),
+                references,
+                issues,
+                mutableSetOf()
+            )
+
+            val packages = collectPackages(references) + rootDocument.getPackages()
+            return SpdxTransitiveDocument(rootDocument, managerName, references, packages, issues)
+        }
+    }
+
+    /**
+     * Get the [SpdxPackage] for the given [identifier] by resolving against packages or external document references
+     * contained in this document. If the package cannot be resolved, add an issue to [issues].
+     */
+    fun getSpdxPackageForId(identifier: String, issues: MutableList<OrtIssue>): SpdxPackage? {
+        return packagesById[identifier]?.also {
+            issueForPackage(identifier)?.let { issues += it }
```

Comment:
IMO the `let` should also be `also` instead as we're not interested in the return value.